src/trackinfoobject.cpp

```diff
@@ -255,7 +255,8 @@ void TrackInfoObject::parse(bool parseCoverArt) {
 
         // Parse the information stored in the sound file.
         Mixxx::TrackMetadata trackMetadata;
-        if (proxy.parseTrackMetadata(&trackMetadata) == OK) {
+        QImage* pCoverArt = parseCoverArt ? &m_coverArt.image : NULL;
+        if (proxy.parseTrackMetadataAndCoverArt(&trackMetadata, pCoverArt) == OK) {
```

Comment:
This took me a while to understand. 
Please do something to help.  
Add a comment or replace this with if/else + two calls for parseTrackMetadataAndCoverArt. 
.. or could we just add two wrappers parseTrackMetadata() and parseCoverArt().